lib/utils_disk.c

```diff
@@ -196,6 +207,28 @@ np_find_name (struct name_list *list, const char *name)
   return FALSE;
 }
 
+/* Returns TRUE if name is in list */
+int
+np_find_regmatch (struct regex_list *list, const char *name)
+{
+  int len;
+  regmatch_t m;
+
+  if (name == NULL) {
+    return FALSE;
+  }
+
+  len = strlen(name);
+
+  for (; list; list = list->next) {
+    if (!regexec(&list->regex, name, 1, &m, 0) && m.rm_so == 0 && m.rm_eo == len) {
```

Comment:
Nope. FAT would still match only FAT. `FAT.*` is another story...

Reply:
ah, I see, you are enforcing a fullmatch here, which I conveniently ignored. Mea culpa.

Reply:
could we please add a little comment, it wasn't obvious to me as well